### PR TITLE
Handle singular and plural verbiage

### PR DESCRIPTION
**Pull request by bsell93**
Hello! Would love to handle singular/plural grammar in the error messages. Open to suggestions & other changes. I thought I'd at least put this out there to see if there are any thoughts on this.

### CODE REVIEW

**Review by scotttrinh — APPROVED** [review]
@bsell93 

Thanks for bringing this up! User facing copy is always such an 🙃 thing in error messaging, so improving that is always a welcome update. There is probably something more consistent that we can do, but I think what you have here is an improvement already, and Colin has stated that copy changes are not "breaking", so I think this is safe to merge. Thanks!

File: src/ZodError.ts [hunk 1]
```diff
@@ -317,11 +317,11 @@ export const defaultErrorMap = (
       if (issue.type === "array")
         message = `Should have ${issue.inclusive ? `at most` : `less than`} ${
           issue.maximum
-        } items`;
+        } item${issue.maximum === 1 ? '' : 's'}`;
       else if (issue.type === "string")
         message = `Should be ${issue.inclusive ? `at most` : `under`} ${
           issue.maximum
-        } characters long`;
+        } character${issue.minimum === 1 ? '' : 's'} long`;
```
Comment (scotttrinh) [inline src/ZodError.ts:324]:
```suggestion
        } character${issue.maximum === 1 ? '' : 's'} long`;
```

File: src/ZodError.ts [hunk 2]
```diff
@@ -302,11 +302,11 @@ export const defaultErrorMap = (
       if (issue.type === "array")
         message = `Should have ${issue.inclusive ? `at least` : `more than`} ${
           issue.minimum
-        } items`;
+        } item${issue.minimum === 1 ? '' : 's'}`;
       else if (issue.type === "string")
         message = `Should be ${issue.inclusive ? `at least` : `over`} ${
           issue.minimum
-        } characters`;
+        } character${issue.minimum === 1 ? '' : 's'} long`;
```
Comment (scotttrinh) [inline src/ZodError.ts:309]: did you mean to add ` long` here? If so, we'll need to update the failing test.

**Review by scotttrinh — CHANGES_REQUESTED** [review]

**Comment by colinhacks** [comment]
Thanks for this PR @bsell93. It motivated me to go a slightly different direction with these error messages that I think is more explicit: https://github.com/colinhacks/zod/commit/2021a5e07c3e0a06a33b4c44441fc9863ae51ae4 

Closing for now.

**Comment by bsell93** [comment]
Glad to bring inspiration! 😁

**Pull request closed without merging**